### ISSUE TITLE
Threads working at one time?

### ISSUE THREAD

**Issue by gregbuchholz** [RReverser/emscripten-rust-test#1]
I've been investigating getting threads working with Rust and wasm32-unknown-emscripten, and I came across a [previous message](https://github.com/emscripten-core/emscripten/issues/14419#issuecomment-857928602) of yours which make it seem like threads were working at one time.  When I try your example program, I get errors seemingly related to thread local storage like:

    ...relocation R_WASM_MEMORY_ADDR_TLS_SLEB cannot be used against non-TLS symbol `std::io::stdio::OUTPUT_CAPTURE::__getit::__KEY::h776cf75763f0fad1`

...can you confirm that threading was working at one time, and there must have been a regression somewhere between now and then?  I've tried with emscripten v3.0.0, and one prior version with the same `wasm-ld` error.

Thanks!

**Comment by RReverser** [comment]
No, what I meant by that comment is that I created a reproduction repo that demonstrates the linking issues (that is, it was broken from the beginning).
